src/Extension/FluentExtension.php

```diff
@@ -679,6 +686,90 @@ public function onAfterDuplicate($original, $doWrite, $relations): void
                 SQLUpdate::create($localisedTable, $fieldsToUpdate, ['RecordID' => $toID])->execute();
             }
         }
+
+        // We need to handle duplicating relations in `localised_copy` into additional locales
+
+        // We don't have any localised relations to cover
+        if (count($copyHasOneRelations) === 0) {
+            return;
+        }
+
+        $currentLocale = FluentState::singleton()->getLocale();
+
+        // We don't have a current locale which indicates that the Fluent setup is incomplete - bail out
+        if (!$currentLocale) {
+            return;
+        }
+
+        $locales = $this->getLocaleCodesForModel($owner);
+
+        // Current locale can be skipped as it was already handled correctly
+        $locales = array_diff($locales, [$currentLocale]);
+
+        // No locales need to be actioned
+        if (count($locales) === 0) {
+            return;
+        }
+
+        $ownerIsVersioned = $owner->hasExtension(Versioned::class);
+
+        foreach ($locales as $locale) {
+            FluentState::singleton()->withState(
+                static function (FluentState $state) use ($owner, $original, $ownerIsVersioned, $locale, $copyHasOneRelations): void {
+                    $state->setLocale($locale);
+
+                    // This is model which was created by the duplication
+                    $localisedOwner = DataObject::get($owner->ClassName)->byID($owner->ID);
+
+                    // This is the model that we were duplicating
+                    $localisedOriginal = DataObject::get($original->ClassName)->byID($original->ID);
```

Comment:
```suggestion
                    // This is localised copy of the record which was created by the duplication
                    $localisedOwner = DataObject::get($owner->ClassName)->byID($owner->ID);

                    // This is the localised copy of the original record that we were duplicating
                    $localisedOriginal = DataObject::get($original->ClassName)->byID($original->ID);
```
This additional context is sort-of self explanatory by the fact we're in `withState()` - but it could be easy to get confused for developers not used to working with fluent or who are just having an off day.